### PR TITLE
Add DSSE processor

### PR DESCRIPTION
**Pull request by mlieberman85**
This implements a processor that unpacks DSSE documents

Signed-off-by: Michael Lieberman <mlieberman85@gmail.com>

### CODE REVIEW

**Comment by mlieberman85** [comment]
A few things that were unclear is whether or not the document tree stuff is intended to handle all the traversal of the tree or if unpack should call unpack on all children.

Another thing, which may be premature optimization is if we don't verify the signature at unpacking time, it means we would need to go back and verify the signature later and cascade the info down the tree

**Comment by lumjjb** [comment]
>A few things that were unclear is whether or not the document tree stuff is intended to handle all the traversal of the tree or if unpack should call unpack on all children.

Only 1 round of unpacking. Since the document types that result from that may not a format they can handle or unknown. The tree construction is handled by the processor code.

>Another thing, which may be premature optimization is if we don't verify the signature at unpacking time, it means we would need to go back and verify the signature later and cascade the info down the tree

Yea - i think that would be a validation step as the ingestor is working through the documents. If it fails a validation step on one of the nodes then it doesn't walk the rest of that branch.

File: pkg/handler/processor/dsse/dsse_test.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		name:      "DSSE Envelope with Unknown Payload",
+		doc:       unknownDSSEDoc,
+		expected:  []*processor.Document{&unpackedUnknownDSSEDoc},
+		expectErr: false,
+	}, {
+		name:      "DSSE Envelope with ITE6",
+		doc:       ite6DSSEDoc,
+		expected:  []*processor.Document{&ite6SLSADoc},
+		expectErr: false,
+	}}
```
Comment (lumjjb) [inline pkg/handler/processor/dsse/dsse_test.go:125]: is there a scenario for DSSE where unpack fails , and unpacks a list of len==0.

File: pkg/handler/processor/dsse/dsse_test.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		Blob:   []byte(ite6SLSA),
+		Type:   processor.DocumentITE6,
+		Format: processor.FormatJSON,
+		SourceInformation: processor.SourceInformation{
+			Collector: "TestCollector",
+			Source:    "TestSource",
+		},
+	}
+)
+
```
Comment (lumjjb) [inline pkg/handler/processor/dsse/dsse_test.go:108]: can we add a simple ValidateSchema test with good schema and bad schema (which should fulfill one case in the unpack tests as well)

**Comment by lumjjb** [comment]
minor requests, else LGTM

File: pkg/handler/processor/dsse/dsse.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+	var doc *processor.Document
+	decodedPayload, err := base64.StdEncoding.DecodeString(envelope.Payload)
+	if err != nil {
+		return nil, err
+	}
+	switch pt := envelope.PayloadType; pt {
+	case string(dsseITE6):
+		doc = &processor.Document{
+			Blob:              decodedPayload,
```
[the remark this answers is not in the document]
Comment (pxp928) [reply pkg/handler/processor/dsse/dsse.go:70]: I see that its stored via the doc tree above. Nevermind.

File: pkg/handler/processor/dsse/dsse_test.go (hunk 1 above)
Comment (mlieberman85) [reply pkg/handler/processor/dsse/dsse_test.go:125]: Will add

File: pkg/handler/processor/dsse/dsse_test.go (hunk 2 above)
Comment (mlieberman85) [reply pkg/handler/processor/dsse/dsse_test.go:108]: Yep, will add

**Review by mihaimaruseac — APPROVED** [review]

**Review by lumjjb — APPROVED** [review]
LGTM

**Pull request merged**